pallets/moonbeam-foreign-assets/src/lib.rs

```diff
@@ -319,6 +329,37 @@ pub mod pallet {
 		pub deposit: BalanceOf<T>,
 	}
 
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub assets: Vec<EvmForeignAssetInfo>,
+		pub _phantom: PhantomData<T>,
+	}
+
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				assets: vec![],
+				_phantom: Default::default(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+		fn build(&self) {
+			for asset in self.assets.clone() {
+				Pallet::<T>::register_foreign_asset(
+					asset.asset_id,
+					asset.xcm_location,
+					asset.decimals,
+					asset.symbol,
+					asset.name,
+				)
+				.expect("couldn't register asset");
```

Comment:
If this is infallible let's rephrase the message as a "should" https://doc.rust-lang.org/std/error/index.html#common-message-styles

Otherwise maybe we could log the error and panic explicitly.

```rust
.unwrap_or_else(|e| {
      log::error!("Failed to register asset {:?}: {:?}", asset.asset_id, e);
      panic!("Genesis build failed: {}", e);
  });
```